src/Entity/PropertyId.php

```diff
@@ -48,7 +48,7 @@ private function assertValidIdFormat( $idSerialization ) {
 	 * @return int
 	 */
 	public function getNumericId() {
-		return (int)substr( $this->serialization, 1 );
+		return (int)substr( $this->getSerialization(), 1 );
```

Comment:
I believe this is wrong, as it should take the last part of the serialized ID and drop `P` from it. E.g. this will break for `foo:P123`.
Also, I believe you forgot to do changes regarding accessing `$this->serialization` you did for this class in `ItemId` class as well.